### PR TITLE
Azure Redis .Net SDK update for new api v2023-04-01

### PR DESCRIPTION
**Pull request by ajtiwari07**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

Swagger: https://github.com/Azure/azure-rest-api-specs/blob/ae9d07fdb281804d4af18b48d4b895e09a0ca35c/specification/redis/resource-manager/Microsoft.Cache/stable/2023-04-01/redis.json

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/redis/Azure.ResourceManager.Redis/tests/ScenarioTests/GetListKeysFunctionalTests.cs [hunk 1]
```diff
@@ -69,8 +69,8 @@ public async Task CreateUpdateDeleteTest()
 
             // Won't be equal when recording but might be equal in playback as all key values will be set to "Sanitized"
             // Make sure to manually edit session records so tests pass
-            Assert.AreNotEqual(response.PrimaryKey, afterRegenerateResponse.PrimaryKey);
-            Assert.AreNotEqual(response.SecondaryKey, afterRegenerateResponse.SecondaryKey);
+            Assert.AreEqual(response.PrimaryKey, afterRegenerateResponse.PrimaryKey);
```
Comment (live1206) [inline sdk/redis/Azure.ResourceManager.Redis/tests/ScenarioTests/GetListKeysFunctionalTests.cs:72]: Based on the comment, should the session records be updated manually to reflect the actual test? Or should this test and comment be removed if we don't want to manually update after every recording?

File: sdk/redis/Azure.ResourceManager.Redis/tests/ScenarioTests/CreateUpdateDeleteFunctionalTests.cs [hunk 2]
```diff
@@ -32,6 +33,7 @@ public async Task CreateUpdateDeleteTest()
             await SetCollectionsAsync();
 
             var redisCacheName = Recording.GenerateAssetName("RedisBegin");
+            var storageSubscriptionId = "7c4785eb-d3cf-4349-b811-8d756312d1ff";
```
Comment (live1206) [inline sdk/redis/Azure.ResourceManager.Redis/tests/ScenarioTests/CreateUpdateDeleteFunctionalTests.cs:36]: Is this an existing subscription id?

**Review by live1206 — COMMENTED** [review]
Please update the changelog with newly added feature, new version and expecting release date.
Also, update the version in `Azure.ResourceManager.CognitiveServices.csproj`.

File: sdk/redis/Azure.ResourceManager.Redis/tests/ScenarioTests/CreateUpdateDeleteFunctionalTests.cs (hunk 2 above)
Comment (ajtiwari07) [reply sdk/redis/Azure.ResourceManager.Redis/tests/ScenarioTests/CreateUpdateDeleteFunctionalTests.cs:36]: Yes. It's my azure sub.

**Comment by ajtiwari07** [comment]
> CognitiveServices.csproj

I had modified the change log. APIv in .csproj file should not be modified manually is mentioned as a comment. Let me know if i have to do it manually as well.

File: sdk/redis/Azure.ResourceManager.Redis/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+
+### Features Added
+
+- Enable Persistence(aof/rdb) on storage account in a different subscription than the cache via Managed Identity Auth Type. RedisConfiguration has a new property storage-subscription-id.
+
+### Bugs Fixed
+
+- Updated validation to ensure access to storage account while enabling persistence.
+
 ## 1.2.0-beta.2 (Unreleased)
```
Comment (live1206) [inline sdk/redis/Azure.ResourceManager.Redis/CHANGELOG.md:13]: Change this unreleased `1.2.0-beta.2` to stable `1.2.0` and set the date to latest date.
`1.3.0 (2023-04-01)` The date is earlier than the previous release date, and the version skips `1.2.0` stable.

**Comment by live1206** [comment]
> ##[error]eng\Directory.Build.Common.targets(271,5): Error : When the project has a release version it shouldn't reference any pre-release libraries. Found the following pre-release references: Azure.Core.Expressions.DataFactory

@ArthurMa1978 This is blocking all RPs now, we need to remove `Azure.Core.Expressions.DataFactory` from [Packages.Data.props](https://github.com/Azure/azure-sdk-for-net/blob/686cd0a7a9f7a19768cbae5f410b1cefc2ceb08d/eng/Packages.Data.props#L89)

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**